src/User.php

```diff
@@ -5067,127 +5065,56 @@ public function showItems($tech)
+                        $linktype .= sprintf(
+                            __('%1$s = %2$s'),
+                            Group::getTypeName(1),
+                            $groups[$data[$field_group]]
                         );
-                        echo "</td><td class='center'>$link</td>";
-                        echo "<td class='center'>";
-                        if (isset($data["serial"]) && !empty($data["serial"])) {
-                             echo $data["serial"];
-                        } else {
-                            echo '&nbsp;';
-                        }
-                        echo "</td><td class='center'>";
-                        if (isset($data["otherserial"]) && !empty($data["otherserial"])) {
-                            echo $data["otherserial"];
-                        } else {
-                            echo '&nbsp;';
-                        }
-                        echo "</td><td class='center'>";
-                        if (isset($data["states_id"])) {
-                            echo Dropdown::getDropdownName("glpi_states", $data['states_id']);
-                        } else {
-                            echo '&nbsp;';
-                        }
-
-                        echo "</td><td class='center'>$linktype</td></tr>";
                     }
+                    $entries[] = [
+                        'itemtype'      => $itemtype,
+                        'id'            => $data["id"],
+                        'type'          => $type_name,
+                        'entity'        => Dropdown::getDropdownName("glpi_entities", $data["entities_id"]),
+                        'name'          => $link,
+                        'serial'        => $data["serial"],
+                        'otherserial'   => $data["otherserial"],
+                        'states'        => Dropdown::getDropdownName("glpi_states", $data['states_id']),
+                        'linktype'      => $linktype,
+                    ];
                 }
             }
-            if ($nb) {
-                echo $header;
-            }
-            echo "</table></div>";
         }
+
+        TemplateRenderer::getInstance()->display('components/datatable.html.twig', [
+            'is_tab'                => true,
+            'items_id'              => $ID,
+            'nofilter'              => true,
+            'columns'               => [
+                'type'          => _n('Type', 'Types', 1),
+                'entity'        => Entity::getTypeName(1),
+                'name'          => __('Name'),
+                'serial'        => __('Serial number'),
+                'otherserial'   => __('Inventory number'),
+                'states'        => __('Status'),
+                'linktype'      => ''
+            ],
+            'entries'               => $entries,
+            'total_number'          => $number,
+            'nopager'               => true,
```

Comment:
Is it intended to hide page and filter controls? It seems like they would be supported.

Reply:
As the result table is the sum of several queries, I thought that managing filters and pagination added dispensable complexity.

But yes it can be managed, what do you think? @orthagh @cedric-anne @trasher

Reply:
After thinking a little to the feature and after the discussion yesterday, we may not be sure someone has thousand of consumables.
This is unlikely but i guess we should explore pagination.
We discussed yesterday about we may query different data source, but it seems you have one query. pagination should feasible. Check src/Item_Process.php file for example 

BTW, I'm not sure filters are required. Could you post a screenshot of how it renders ?

Reply:
Sorry, there is an ambiguity.
It's not in the consumable tab, but in the items tab (I didn't put it in the original comment, I just corrected it).
The last commit adds pagination.